flowz/channels/core.py

```diff
@@ -401,9 +401,13 @@ def __reader__(self, thischan):
             set_channel_done_exception(head, "ReadChannel.__reader__")
         except:
             # Capture exception information, including traceback
-            # TODO Possible issue when Python3-compatibility is important
-            head.set_exc_info(sys.exc_info())
-
+            exc_info = sys.exc_info()
```

Comment:
Since this occurs in at least a couple of places, how would you feel about making a utility function that does this tornado-compatibility thing?